mso/datasource_mso_schema_site_bd.go

```diff
@@ -66,21 +70,23 @@ func dataSourceMSOSchemaSiteBdRead(d *schema.ResourceData, m interface{}) error
 		return fmt.Errorf("No Sites found")
 	}
 	stateSite := d.Get("site_id").(string)
+	stateTemplate := d.Get("template_name").(string)
 	found := false
 	statebd := d.Get("bd_name").(string)
-	for i := 0; i < count; i++ {
+	for i := 0; i < count && !found; i++ {
 		tempCont, err := cont.ArrayElement(i, "sites")
 		if err != nil {
 			return err
 		}
 		apiSite := models.StripQuotes(tempCont.S("siteId").String())
+		apiTemplate := models.StripQuotes(tempCont.S("templateName").String())
 
-		if apiSite == stateSite {
+		if apiSite == stateSite && apiTemplate == stateTemplate {
 			bdCount, err := tempCont.ArrayCount("bds")
 			if err != nil {
 				return fmt.Errorf("Unable to get bd list")
 			}
-			for j := 0; j < bdCount; j++ {
+			for j := 0; j < bdCount && !found; j++ {
```

Comment:
Do we need this !found here? Is the break after found not already breaking the for loop?

Reply:
ok